src/main/kotlin/org/jitsi/nlj/stats/PacketDelayStats.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.nlj.stats
+
+import org.jitsi.nlj.PacketInfo
+import java.util.concurrent.atomic.AtomicLong
+
+class DelayStats {
+    private val totalDelayMs = AtomicLong(0)
+    private val totalPackets = AtomicLong(0)
+    val averageDelay: Double
+        get() = totalDelayMs.get() / totalPackets.get().toDouble()
+    val maxDelayMs = AtomicLong(0)
+
+    fun addPacket(packetInfo: PacketInfo) {
```

Comment:
I think that `DelayStats` should just work with delays and not care about `PacketInfo`. It would be easier to re-use for other things as well.

Reply:
what else would we use it for? are you thinking like method calls or other tasks?

Reply:
Tracking the delay of just the receive pipeline, or just the send pipeline, or of the packets in a specific queue. It's a cliché, but: avoid unnecessary tight coupling